rust/kernel/irq.rs

```diff
@@ -9,6 +9,7 @@
 
 #![allow(dead_code)]
 
+use crate::prelude::*;
```

Comment:
I think it should generally be fine to include prelude. I could remove the `Error` and `Result` import so that this import is used more than just `vtable`.

Reply:
Even if you want to include `prelude::*`, you should add it to the line below for consistency.

Now, given the choice of adding `macros::vtable` or `prelude::*`, I think the former is preferable because we don't pollute the namespace with more symbols than needed. (prelude is meant for drivers.)

You're obviously welcome to disagree with our handling or `prelude`, but in that case I'd argue that you're unnecessarily conflating issues: let's open another PR or issue to discuss that if you'd like. I'm ready to merge this once we come to an agreement on the `use` clause.